tetris.py

```diff
@@ -192,5 +196,16 @@ def block_rotator(self, event):
             rotate.append((y, x))
         self.current_block = rotate
 
+    def full_line_clear(self):
+        y_coordinates = [y for (x, y) in self.landed_blocks]
+        coordinates_counter = collections.Counter(y_coordinates)
+        print(coordinates_counter)
+        for y_line, count in coordinates_counter.items():
```

Comment:
The counter is not necessarily in the correct order:

```
>>> collections.Counter([(0, 0), (0, 1)])
Counter({(0, 0): 1, (0, 1): 1})
>>> collections.Counter([(0, 1), (0, 0)])
Counter({(0, 1): 1, (0, 0): 1})
```

Notice how in these examples, the coordinate with bigger `y` can be first or second in the counter, depending on the order of the list passed in. In this case, the order of the list depends on how the blocks happen to fall down, and can't be controlled.

To fix this, you can loop over `range(height)` which is always in increasing order, or `reversed(range(height))` which is always in decreasing order:

```suggestion
        for y_line in range(game_height):   # TODO: should this be reversed?
            count = coordinates_counter[y_line]
```

Reply:
Sometimes the order happens to be correct, and multiple full lines are deleted properly. But sometimes the order is wrong and only one full line is deleted.